### PR TITLE
update twitter meta info 

### PR DESCRIPTION
**Pull request by NotJohs**
re [#2206](https://github.com/opencollective/opencollective/issues/2206)
pr to [#2093](https://github.com/opencollective/opencollective-frontend/pull/2093)

The size of a card depends on description and image size.

**no image for root URL**

### CODE REVIEW

**Comment by znarf** [comment]
@NotJohs why have you decided to change the image size?

This is the current result from the branch.

Home Page

Capture d’écran 2019-07-12 à 09 15 49

Collective Page

Capture d’écran 2019-07-12 à 09 16 28

@xdamman does that work for you? any feedback?

**Comment by NotJohs** [comment]
@znarf https://github.com/opencollective/opencollective-frontend/pull/2093

**Comment by znarf** [comment]
@NotJohs I don't see any feedback on the image size. Did I miss something?

**Comment by NotJohs** [comment]
@znarf  For the root dir the there is no image provided.

File: src/components/Header.js [hunk 1]
```diff
@@ -22,15 +22,17 @@ class Header extends React.Component {
     super(props);
     const { description, image, twitterHandle, title } = props;
     const meta = {
-      'twitter:site': 'opencollect',
-      'twitter:creator': twitterHandle,
+      'twitter:site': '@opencollect',
+      'twitter:creator': twitterHandle ? `@${twitterHandle}` : ' ',
```
Comment (xdamman) [inline src/components/Header.js]: why a non empty string?

Comment (NotJohs) [reply src/components/Header.js]: At the root dir, it returns undefined.

Comment (xdamman) [reply src/components/Header.js]: yes an empty string but why a space? Return `''`, not `' '`;

**Comment by xdamman** [comment]
Can we fall back to https://opencollective.com/static/images/opencollectivelogo-480x80.png
if no image is provided?

**Comment by NotJohs** [comment]
@xdamman sure.

**Comment by xdamman** [comment]
Or https://opencollective.com/static/images/opencollectiveicon-240x240@2x.png
if you think that we need a square one

**Comment by NotJohs** [comment]
@xdamman Alright. Will do.

**Comment by NotJohs** [comment]
@xdamman  Will use x240

**Comment by znarf** [comment]
@NotJohs you modified "summary_large_image". is there any reason?

**Comment by NotJohs** [comment]
@znarf 
[optimizations](https://developer.twitter.com/en/docs/tweets/optimize-with-cards/overview/summary-card-with-large-image) for using the twitter card with a larger image

**Comment by znarf** [comment]
@NotJohs you're not answering the question, can you explain why *you decided* to do that?

**Comment by NotJohs** [comment]
@znarf 
i used summary_large_image because:
- to give the twitter card a bolder look and better overall design.
- xdamman stated it was a regression how the card was looking  https://github.com/opencollective/opencollective-frontend/pull/2093

**Comment by NotJohs** [comment]
@znarf sorry I missed the part where you mentioned image size, I thought you were asking about the lack of image on open collective

File: src/components/Header.js (hunk 1 above)
Comment (NotJohs) [reply src/components/Header.js]: will fix

File: src/components/Header.js [hunk 2]
```diff
@@ -22,15 +22,17 @@ class Header extends React.Component {
...
       description: truncate(description, 256),
-      'twitter:card': 'summary',
-      'twitter:title': `${title} - Open Collective`,
-      'twitter:description': description,
-      'twitter:image': image,
+      'og:description': truncate(description, 256),
+      'twitter:card': 'summary_large_image',
+      'twitter:title': title ? `${title} - Open Collective` : 'Open Collective',
+      'twitter:description': truncate(description, 256),
+      'twitter:image': image ? image : 'https://opencollective.com/static/images/opencollectiveicon-240x240@2x.png',
```
Comment (xdamman) [inline src/components/Header.js]: it's usually not a good practice to repeat the same string twice.
Keep it in a variable, e.g. `defaultImage = ...`

Comment (NotJohs) [reply src/components/Header.js]: I understand

**Comment by NotJohs** [comment]
Screenshot 2019-07-12 at 11 55 58
Is this alright?

File: src/components/Header.js [hunk 3]
```diff
@@ -21,16 +21,20 @@ class Header extends React.Component {
   constructor(props) {
     super(props);
     const { description, image, twitterHandle, title } = props;
+    const defaultTitle = title ? `${title} - Open Collective` : 'Open Collective';
+    const defaultImage = image ? image : 'https://opencollective.com/static/images/opencollectiveicon-240x240@2x.png';
```
Comment (xdamman) [inline src/components/Header.js]: Huh. That's not the right variable name if you use it this way. 
If you name it `defaultImage`, then it should be a constant that is independent of any condition.

```js
const defaultImage = 'https://opencollective.com/static/images/opencollectiveicon-240x240@2x.png';
meta['twitter:image'] = image ? image : defaultImage;
meta['twitter:image'] = image || defaultImage; // better
```

Comment (NotJohs) [reply src/components/Header.js]: Understood

Comment (NotJohs) [reply src/components/Header.js]: Screenshot 2019-07-12 at 16 58 59

is this alright?

Comment (xdamman) [reply src/components/Header.js]: Much much better! :-)

Comment (NotJohs) [reply src/components/Header.js]: Alright, would push.

**Comment by znarf** [comment]
@NotJohs I'm confused, why have you closed this Pull Request?

**Comment by NotJohs** [comment]
@znarf 
I made a mess and I didn't want to disturb u with rebasing, so I cleaned up my branch and made a new PR

PR https://github.com/opencollective/opencollective-frontend/pull/2157

**Pull request closed without merging**